Create 2024-07-08-geoserver-tutorials.md (#203)
geoserver/geoserver.github.io: _posts/2024-07-08-geoserver-tutorials.md

---
author: Nima Ghasemloo
layout: post
title: Utilizing the Demo Section in Geoserver
date: 2024-07-08
categories:   
- Tutorials
---

[GeoSpatial Techno](https://www.youtube.com/@geospatialtechno) is a startup focused on geospatial information that is providing e-learning courses to enhance the knowledge of geospatial information users, students, and other startups. The main approach of this startup is providing quality, valid specialized training in the field of geospatial information.

( [YouTube](https://www.youtube.com/@geospatialtechno)
| [LinkedIn](https://www.linkedin.com/in/geospatialtechno)
| [Facebook](https://www.facebook.com/geospatialtechno)
| [X](https://twitter.com/geospatialtechn)
)

----

### Utilizing the Demo Section in Geoserver
In this session we aim to provide content on how to use the **Demo menu and its modules in GeoServer** and teach the process of making different requests and getting responses from the server. If you want to access the complete tutorial, simply click on the [link](https://youtu.be/3xKKI8irlQk).

> [![](https://img.youtube.com/vi/3xKKI8irlQk/0.jpg)](https://www.youtube.com/watch?v=3xKKI8irlQk)

Reviewer: Just glancing over this video, I noticed it's not a match for the current demo request builder anymore, it has been rewritten starting with GeoServer 2.25.2 and 2.24.4 (which we also indicated as mandatory upgrades for production systems, due to the recent RCE vulnerability). 

Differences are small, but some indication should be provided that the video showcases an outdated GeoServer version.

Author: Hi, As you said, the differences are small, but at the beginning of the blog post, I suggest using GeoServer versions 2.24.x and 2.25.x.

Reviewer 2: I did tell ya that it was being updated :) The workflow is slightly different, and I believe more clear.

Reviewer 2: Q: Do you have capacity to re-record the video with the improved screens workflow? It would be excellent timing ...

Author: We have recorded videos for demo sections, types of WFS filtering, transactions, WPS installation, and spatial analysis. Next, we will create a video about upgrading GeoServer to the latest version and using it.

Reviewer 2: I was asking specifically about the "utilizing demos" section; even adding to the video to cover the new changes would be welcome. 

We do not change the user experience very often so your timing here is bad luck

Reviewer 2: oooh - updating video would be very welcome: https://docs.geoserver.org/latest/en/user/installation/upgrade.html

Things I will check in review:
1. move to external data directory (have to discuss GEOSERVER_DATA_DIR setting options)
2. make a list of extensions installed
3. backup data directory 1st 
4. update binary or war as appropriate, and install extensions

It may help to show updating and forgetting to install an extension like WPS, and the resulting failure.

Reviewer 2: @NimaGhasemloo was able to answer my question in chat, the video is already prepared and will not be updated for the new user interface at this time.

---
author: Nima Ghasemloo
layout: post
title: Utilizing the Demo Section in Geoserver
date: 2024-07-08
categories:   
- Tutorials
---

[GeoSpatial Techno](https://www.youtube.com/@geospatialtechno) is a startup focused on geospatial information that is providing e-learning courses to enhance the knowledge of geospatial information users, students, and other startups. The main approach of this startup is providing quality, valid specialized training in the field of geospatial information.

( [YouTube](https://www.youtube.com/@geospatialtechno)
| [LinkedIn](https://www.linkedin.com/in/geospatialtechno)
| [Facebook](https://www.facebook.com/geospatialtechno)
| [X](https://twitter.com/geospatialtechn)
)

----

### Utilizing the Demo Section in Geoserver
In this session we aim to provide content on how to use the **Demo menu and its modules in GeoServer** and teach the process of making different requests and getting responses from the server. If you want to access the complete tutorial, simply click on the [link](https://youtu.be/3xKKI8irlQk).

[![](https://img.youtube.com/vi/3xKKI8irlQk/0.jpg)](https://www.youtube.com/watch?v=3xKKI8irlQk)

## Introduction

> The Demo application is a valuable tool to help you practice with requests. It’s used to create and test requests from the GeoServer. The user can see the response to the requests and solve the problems. To become familiar with the demo section, scroll down and click on the "Demo" link. This page contains these options:

Reviewer 2: ```suggestion
The Demo page provides valuable tool to help you practice with requests and understand how GeoServer works. The **Demo Requests** is used to create and test requests to GeoServer. The user can see the response to successful requests, or troubleshoot problems. To become familiar with the demo page, scroll down and click on the **Demo** link. This page contains these options:
```

Reviewer 2: It is not so much a demo application as a page providing several tools to try out geoserver. the SRS list and other tools do not do requests at all, only demonstrate how geoserver works?

Author: Done.

- Demo Requests
- SRS List
- Reprojection Console
- WCS Request Builder.

If you have installed the WPS extension, you will see an additional option called WPS Request Builder. 

> **Note.** You do not need to be logged into GeoServer to access this page.

Reviewer 2: ```suggestion
**Note.** These tools are for the public like **Layer Preview**, you do not need to be logged into GeoServer to access this page.
```

Author: Done.

Now let's explore some basic operations:
- From the drop-down list, you can select a set of prepared requests. They are listed with a syntax declaring the standard as a prefix and the standard's version as a suffix. Choose WFS_getCapabilities-1.1.xml.
- Press the Submit button. A new panel is shown, and, after a while, it lists the XML response from GeoServer.
- Another basic WFS operation is getFeature, which will retrieve a feature for you. Select WFS_getFeature-1.0.xml. If you look at the XML code, you can see a clear reference to the topp:states layer, which is included in the sample set.
- Press the Submit button. A new panel is shown, and, after a while, it lists the XML response from GeoServer. The code is a GML representation of the features with `fid = 3`, as requested in the filter.

The Demo requests interface lets you select sample requests and modify them to perform testing on GeoServer. When in doubt about a specific operation, this application should be the first point where you go to debug. From here, you can concentrate on the request's syntax, avoiding network issues or other problems that you may have experienced with an external client.

> 

Reviewer 2: ```suggestion

**New Feature:** For GeoServer 2.25.2 the Demo Request page has been improved to show response Headers, and provide the option to pretty print XML output.
```

Author: Done.

- WCS Version—Version of WCS to use when crafting the request.
- Coverage name—Coverage to use in the request.

**Note.** All other options displayed will be non-functional until Coverage name is selected. Once selected, the remainder of the form will be displayed. Watch the video for a full explanation of these options: 
- Spatial subset
- Coordinate reference system
- Specify source grid manually (1.0.0 only)
- Target coverage layout (1.1.1 only)
- Target CRS
- Output format

There is also a link for Describe coverage next to the Coverage name which will execute a WCS DescribeCoverage request for the particular layer. At the bottom of the form are two buttons for form submission:
- Get Coverage: It executes a GetCoverage request using the parameters in the form.
- Generate GetCoverage XML: Clicking this button generates the GetCoverage request based on the form parameters.

By using this generated XML code, you can easily construct and customize your GetCoverage requests with various parameters, making it easier to retrieve the desired coverage data from GeoServer. 

> 

Author: Done.

## WPS Request Builder
GeoServer with the WPS extension installed includes a request builder for generating and executing WPS processes. Using this tool can greatly simplify the process of authoring WPS requests and making your work with WPS much more convenient and efficient. It's always nice to have assistance in tasks that can be complex or time-consuming. This tool can be a valuable asset in your WPS workflow.

In future sessions, we'll dive deep into the Web Processing Service or WPS and explore its functionalities and how to use different functions in detail. So get ready to expand your knowledge and learn how to leverage the power of WPS for your geospatial analysis needs!

> 

Reviewer 2: ```suggestion

**New Feature:** For GeoServer 2.25.2 the WPS Request Builder has the new option to open the generated request in the **Demo Request Builder**.
```

Author: Done.
